haystack/nodes/question_generator/question_generator.py

```diff
@@ -93,7 +98,7 @@ def __init__(
         self.split_overlap = split_overlap
         self.preprocessor = PreProcessor()
         self.prompt = prompt
-        self.num_queries_per_doc = num_queries_per_doc
+        self.num_queries_per_doc = max(num_queries_per_doc, 3)
```

Comment:
To cap the value to 3 I think you need to use `min()` here and not `max()`. Using `max` tells us that we will always generate at least 3 queries.

Reply:
Wow, what nonsense! Thanks @sjrl